source/mainnet/smart-contracts/onboarding-guide-ethereum-developers/faq.rst

```diff
@@ -0,0 +1,792 @@
+    get an error message with additional information on what JSON object format is expected.
+    This information can help you to create the correct JSON object.
+
+    .. image:: ./images/onboarding_ethereum_developers_7.png
+        :width: 100 %
+
+    **Example 2 (Schema is embedded into the smart contract and the `parameter-json`
+    flag provides the input parameters as a JSON object; the return value is displayed
+    in a human-readable format due to the provided additional information in the schema)**
+
+    .. image:: ./images/onboarding_ethereum_developers_8.png
+        :width: 100 %
+
+    **Example 3 (Schema is provided as a separate file and the `parameter-json` flag
+    provides the input parameters as a JSON object; the return value is displayed in a
+    human-readable format due to the provided additional information in the schema)**
+
+    .. image:: ./images/onboarding_ethereum_developers_9.png
+        :width: 100 %
+
+    **Example 4 (No schema is provided and the `parameter-binary` flag provides the input
+    parameters as a raw bytes string; the return value is displayed in raw bytes because no additional
+    information is available on how to represent the bytes in a human-readable format)**
+
+    .. image:: ./images/onboarding_ethereum_developers_10.png
+        :width: 100 %
+
+.. dropdown:: How can I create the parameter data if I want to use binary input instead of a JSON input?
+
+    When using the ``concordium-client`` to interact with smart contracts the input and output parameters
+    can be either in human-readable format (with a schema) or in raw bytes (binary format).
+    If you want to use the binary format, the below command shows that a `myInputParameters.bin` file is required.
+
+    .. code-block:: console
+
+        $concordium-client contract update <ContractIndex> --entrypoint <ContractEntryPoint> --parameter-binary myInputParameters.bin --sender <Account> --energy 12345678
+
+    You can create such a `myInputParameters.bin` file by adding the below lines to your test cases replacing the `ExampleParams` struct with your input parameter struct for that function.
+
+    .. code-block:: rust
+
+        let parameter = ExampleParams {
+            example_key1: value1,
+            example_key1: value2,
+        }
+        let parameter_bytes = to_bytes(&parameter);
+        std::fs::write("myInputParameters.bin", &parameter_bytes).expect("Failed to write parameter file");
+
+    When running the tests with the below command the `myInputParameters.bin` file is created in the current folder.
+
+    .. code-block:: console
+
+        $cargo test
+
+Events:
+=======
+
+.. dropdown::  Where can I find a logged event on testnet/mainnet?
+
+    You can look up a transaction hash on the `dashboard <https://dashboard.testnet.concordium.com/lookup/13ded9aaf6085e970b2cf3874431de5805ffa35a553c93707d1863a8888e8aa4>`_.
```

Comment:
I think this should link to the main page of the dashboard, and not the specific transaction.

Reply:
The next sentence talks about `the full execution chain` that is visible. If people just open the link (without entering a specific txHash), they might wonder what the next sentences are talking about. I changed the above sentence to make it clearer that the link opens up a specific txHash that is used as an example. 

I changed the sentence to:

You can look up a transaction hash on the dashboard such as `this example <https://dashboard.testnet.concordium.com/lookup/13ded9aaf6085e970b2cf3874431de5805ffa35a553c93707d1863a8888e8aa4>`_.